### DISCUSSION TITLE
Canvas background always white?

### DISCUSSION THREAD

**Discussion by jacopsd** [GrapesJS/grapesjs#5451] [Q&A] [answered]
Loading a page with a transparent background gives unexpected results since grapesjs always inserts a white background for the canvas.

This code is seen starting with the grapesjs iframe tag:
<img width="636" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/210014/ae1a1682-1f26-4c29-ab7a-12aef92f4973">

Can this background-color style rule be removed or changed somewhere pls?

**Comment by javadsamiee** [comment]
You can change it with _baseCss_ in the init

`baseCss: 'body { background: red; }',`

**Reply by jacopsd** [reply]
Thank you. Indeed, this answers the color part of the question. (wonder where this is documented?)

Still for the transparency: when using rgba with a transparency value like:
`baseCss: 'body { background:rgba(86,35,255,0.6); }',`

... there is still a white opaque background in the canvas "behind" this rgba background. This seems to make it impossible to make the page really transparent.

**Reply by javadsamiee** [reply]
I found the editor has a background that your transparent has been mixed with it.

<img width="1492" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/65819265/ae8e6504-e6e0-43d4-b717-e67a1eae2abd">

**Reply by jacopsd** [reply]
Still this doesn't make the canvas transparent: 
`body, .gjs-cv-canvas, .gjs-cv-canvas-bg { background:rgba(86,35,255,0.6)!important; }`

**Reply by javadsamiee** [reply]
Can you change alpha to 0 and tell me the background color behind it? white?

`background:rgba(86,35,255,0)`

If you remove all background colors, you will finally have a white background for the browser that you can not remove.

**Reply by jacopsd** [reply]
Same result.
On the containing page where I open the grapesjs editor, there is a background picture. Target is to see that picture "though" the page being edited with the grapesjs editor. Still the grapesjs canvas is not transparent.

**Reply by javadsamiee** [reply]
Are you able to create a reproducible demo (by using for example [stackblitz](https://stackblitz.com/))?

**Reply by jacopsd** [reply]
Here is a jsfidl. It still shows an error on the console (not sure why), but it shows the background behaviour:

https://jsfiddle.net/jacopsd/wxht8veu/3/

**Reply by javadsamiee** [reply, marked as answer]
Check https://jsfiddle.net/9t4zkaqp/11/

I added a custom style and modified the baseCss

`baseCss: 'body { background: transparent !important; }',`

```
<style>
.gjs-editor { background:rgba(86,35,255,0.6) !important;  }
</style>
```

**Reply by jacopsd** [reply]
Indeed, this solves the question. Thank you.
